plugins/inputs/elasticsearch_query/aggregation_parser.go

```diff
@@ -0,0 +1,117 @@
+package elasticsearch_query
+
+import elastic "gopkg.in/olivere/elastic.v5"
+
+func (e *ElasticsearchQuery) parseAggregationResult(aggregationQueryList *[]aggregationQueryData, searchResult *elastic.SearchResult) error {
+
+	var measurements = make(map[string][]*aggKey)
+
+	for _, aggregationQuery := range *aggregationQueryList {
+		m := measurements[aggregationQuery.measurement]
+		a := aggKey{function: aggregationQuery.function, field: aggregationQuery.field, name: aggregationQuery.name}
+
+		if m == nil {
+			m = []*aggKey{&a}
+		} else {
+			m = append(m, &a)
+		}
+		measurements[aggregationQuery.measurement] = m
+	}
+
+	// recurse over aggregation results
+	for measurementName, aggKey := range measurements {
+		fields := make(map[string]interface{})
+		tags := make(map[string]string)
+
+		err := e.recurseResponse(aggKey, searchResult.Aggregations, measurementName, fields, tags)
```

Comment:
Please put a comment here that you need to recursively resolve the fields and tags as otherwise it's strange to pass the data-structures down without using them afterwards.